Tests/OpenTelemetryTests/Processing/BatchSpanProcessorTests.swift

```diff
@@ -0,0 +1,160 @@
+            processor.processEndedSpan(span)
+        }
+
+        try exporter.shutdownGracefully().wait()
+
+        XCTAssertEqual(exporter.spans.count, 3, "All 3 spans should have been exported.")
+        XCTAssertEqual(exporter.numberOfBatches, 1, "All 3 spans should have been exported in one batch.")
+
+        XCTAssertEqual(exporter.spans[0].context.spanID, span1.context.spanID)
+        XCTAssertEqual(exporter.spans[1].context.spanID, span2.context.spanID)
+        XCTAssertEqual(exporter.spans[2].context.spanID, span3.context.spanID)
+
+        try processor.shutdownGracefully().wait()
+    }
+
+    func test_exportsMultipleBatches() throws {
+        let span1 = try recordedSpan()
+        let span2 = try recordedSpan()
+        let span3 = try recordedSpan()
+        let span4 = try recordedSpan()
+        let span5 = try recordedSpan()
+        let span6 = try recordedSpan()
+        let span7 = try recordedSpan()
+
+        let exporter = AwaitingSpanExporter(expectedNumberOfSpans: 7, eventLoopGroup: eventLoopGroup)
+        let processor = OTel.BatchSpanProcessor(
+            maxBatchSize: 3,
+            interval: .zero,
+            exportingTo: exporter,
+            eventLoopGroup: eventLoopGroup
+        )
+
+        for span in [span1, span2, span3, span4, span5, span6, span7] {
+            processor.processEndedSpan(span)
+        }
+
+        try exporter.shutdownGracefully().wait()
+
+        XCTAssertEqual(exporter.spans.count, 7, "All 7 spans should have been exported.")
+        XCTAssertEqual(exporter.numberOfBatches, 3, "The 7 spans should have been exported in 3 batches.")
+
+        XCTAssertEqual(exporter.spans[0].context.spanID, span1.context.spanID)
+        XCTAssertEqual(exporter.spans[1].context.spanID, span2.context.spanID)
+        XCTAssertEqual(exporter.spans[2].context.spanID, span3.context.spanID)
+        XCTAssertEqual(exporter.spans[3].context.spanID, span4.context.spanID)
+        XCTAssertEqual(exporter.spans[4].context.spanID, span5.context.spanID)
+        XCTAssertEqual(exporter.spans[5].context.spanID, span6.context.spanID)
+        XCTAssertEqual(exporter.spans[6].context.spanID, span7.context.spanID)
+
+        try processor.shutdownGracefully().wait()
+    }
+
+    func test_dropsOldestSpansWhenQueueIsFull() throws {
+        let span1 = try recordedSpan()
+        let span2 = try recordedSpan()
+        let span3 = try recordedSpan()
+        let span4 = try recordedSpan()
+
+        let exporter = AwaitingSpanExporter(expectedNumberOfSpans: 3, eventLoopGroup: eventLoopGroup)
+        let processor = OTel.BatchSpanProcessor(
+            maxBatchSize: 3,
+            maxQueueSize: 3,
+            interval: .milliseconds(10),
+            exportingTo: exporter,
+            eventLoopGroup: eventLoopGroup
+        )
+
+        for span in [span1, span2, span3, span4] {
+            processor.processEndedSpan(span)
+        }
+
+        try exporter.shutdownGracefully().wait()
```

Comment:
is there also a forcefully shutdown btw?

Reply:
Not that I know of. There's a `forceFlush` though which I haven't added to `OTelSpanProcessor` yet. I'll add an issue to revisit this in the future.